clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -1111,6 +1112,87 @@ public void testSubscriptionChangesWithAutoCommitDisabled() {
         consumer.close();
     }
 
+    @Test
+    public void testUnsubscribeWillTriggerPartitionsRevokeWithValidGeneration() {
+        Time time = new MockTime();
+        SubscriptionState subscription = new SubscriptionState(new LogContext(), OffsetResetStrategy.EARLIEST);
+        ConsumerMetadata metadata = createMetadata(subscription);
+        MockClient client = new MockClient(time, metadata);
+
+        initMetadata(client, Collections.singletonMap(topic, 1));
+        Node node = metadata.fetch().nodes().get(0);
+
+        ConsumerPartitionAssignor assignor = new RangeAssignor();
+        KafkaConsumer<String, String> consumer = newConsumer(time, client, subscription, metadata, assignor, false, groupInstanceId);
+
+        initializeSubscription(consumer, getExceptionConsumerRebalanceListener());
+
+        prepareRebalance(client, node, assignor, singletonList(tp0), null);
+
+        try {
+            consumer.updateAssignmentMetadataIfNeeded(time.timer(Long.MAX_VALUE));
+            consumer.poll(Duration.ZERO);
+            fail("Should throw exception");
+        } catch (Throwable e) {
+            assertEquals(partitionAssigned + singleTopicPartition, e.getCause().getMessage());
+        }
+
+        try {
+            consumer.unsubscribe();
+            fail("Should throw exception");
+        } catch (Throwable e) {
+            assertEquals(partitionRevoked + singleTopicPartition , e.getCause().getMessage());
+        }
+    }
+
+    @Test
+    public void testUnsubscribeWillTriggerPartitionsLostWithNoGeneration() throws InterruptedException {
```

Comment:
nit: `unsubscribeShouldTrigger...`

Simplify to `throws Exception` -- it does not matter for a test which exception it throws.

Reply:
Interesting, never heard of this before.

Reply:
It does not matter too much, it just a question of code style (that we cannot enforce via checkstyle AFAIK)